components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3429,11 +3448,25 @@ std::string RewardsServiceImpl::GetExternalWalletType() const {
           country_codes::CountryCharsToCountryID(country.at(0), country.at(1));
 
       if (id == current_country)
-        return ledger::constant::kWalletBitflyer;
+        return true;
     }
   }
+  return false;
+}
 
+std::string RewardsServiceImpl::GetExternalWalletType() const {
+  if (IsBitFlyerRegion()) {
+    return ledger::constant::kWalletBitflyer;
+  }
+#if defined(OS_ANDROID)
   return ledger::constant::kWalletUphold;
+#endif
+
+  return profile_->GetPrefs()->GetString(prefs::kExternalWalletType);
+}
+
+void RewardsServiceImpl::SetExternalWalletType(const std::string wallet_type) {
```

Comment:
This needs validation for the input value, it would be better as an enum that gets translated to a string in prefs

Reply:
also this is a problem because GetExternalWalletType ignores this value in some cases, but there's no indication of that for callers to `SetExternalWalletType`

Reply:
also the flag for gemini appears to be ignored here and the getter so what happens if you enable it, set it to gemini and then disable it?

Reply:
resolved: `SetExternalWalletType` checks if the `wallet_type` is the valid provider list before updating the preference.

Reply:
but we're still using a string which is really fragile and unreliable. An enum translated to a string will have a case statement with a default NOTREACHED to ensure we're passing a valid value